src/controllers/post.js

```diff
@@ -1,15 +1,26 @@
-import { sendDataResponse } from '../utils/responses.js'
+import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
 import { PrismaClient } from '@prisma/client'
 const prisma = new PrismaClient()
 
 export const create = async (req, res) => {
   const { content } = req.body
+  const { id } = req.user
 
   if (!content) {
-    return sendDataResponse(res, 400, { content: 'Must provide content' })
+    return sendMessageResponse(res, 400, 'Must provide content')
   }
 
-  return sendDataResponse(res, 201, { post: { id: 1, content } })
+  try {
+    const createdPost = await prisma.post.create({
+      data: {
+        content,
+        userId: id
+      }
+    })
+    return sendDataResponse(res, 201, { post: createdPost })
+  } catch (err) {
+    return sendMessageResponse(res, 500, 'Unable to create post')
```

Comment:
Almost. You still want to throw the error after sending the response to the client so the error handling middleware can log the error to the dev console, it helps us diagnose errors in the live app.

```js
} catch (err) {
  sendMessageResponse(res, 500, 'Unable to create post')
  throw err
}
```